packages/dapp-high-roller/src/components/app-game/app-game.tsx

```diff
@@ -62,27 +208,58 @@ export class AppGame {
   }
 
   async handleRoll(): Promise<void> {
-    await Promise.all([
-      this.animateRoll("myRoll"),
-      this.animateRoll("opponentRoll")
-    ]);
-
-    this.myRoll = this.generateRoll();
-    this.opponentRoll = this.generateRoll();
-    const totalMyRoll = this.myRoll[0] + this.myRoll[1];
-    const totalOpponentRoll = this.opponentRoll[0] + this.opponentRoll[1];
-    if (totalMyRoll > totalOpponentRoll) {
-      this.myScore += 1;
-      this.gameState = GameState.Won;
-    } else if (totalMyRoll < totalOpponentRoll) {
-      this.opponentScore += 1;
-      this.gameState = GameState.Lost;
-    } else {
-      this.gameState = GameState.Tie;
+    // this.appInstance = this.cfProvider.appInstances[this.appInstanceId]; // this works but is private/not recommended
+    console.log(this.appInstance); // this should be the injected appInstance from the Tunnel
+
+    if (this.isProposing) {
+      if (this.highRollerState.stage === HighRollerStage.PRE_GAME) {
+        await Promise.all([
+          this.animateRoll("myRoll"),
+          this.animateRoll("opponentRoll")
+        ]);
+        const startGameAction: Action = {
+          number: 0,
+          actionType: ActionType.START_GAME,
+          actionHash: nullValueBytes32
+        };
+        this.highRollerState = await takeAction(startGameAction);
+        const numberSalt =
+          "0xdfdaa4d168f0be935a1e1d12b555995bc5ea67bd33fce1bc5be0a1e0a381fc90";
+        const playerFirstNumber = Math.floor(Math.random() * Math.floor(1000));
+        const hash = computeCommitHash(numberSalt, playerFirstNumber);
+
+        const commitHashAction: Action = {
+          number: playerFirstNumber,
+          actionType: ActionType.COMMIT_TO_HASH,
+          actionHash: hash
+        };
+        this.highRollerState = await takeAction(commitHashAction);
+        onUpdateState().then((state: HighRollerAppState) => {
```

Comment:
this isn't right. the `updateState` event is handled in `app-provider` so this wouldn't be here